### PR TITLE
Echo server dev

### PR DESCRIPTION
**Pull request by nadiacollado**

### CODE REVIEW

File: src/main/java/echoserver/EchoClient.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package echoserver;
+import java.io.*;
+import java.net.*;
+
+public class EchoClient {
+    private static int port = 8080;
+    private static String host;
+    private static Socket clientSocket;
+
+    public static void main(String[] args) {
```
Comment (nadiacollado) [inline src/main/java/echoserver/EchoClient.java:10]: Does the EchoClient need its own run task/cmd? Or are we only ever interacting with it through the EchoServer run task?

File: src/main/java/echoserver/EchoServer.java [hunk 2]
```diff
@@ -5,92 +5,37 @@
...
         try {
-            serverSocket = startServerSocket(port);
-            System.out.println("Listening for connection on port " + port);
-
-            while (true) {
-                Socket clientSocket = startClientSocket(serverSocket);
-                startSocketIO(clientSocket);
-            }
+            ServerSocketWrapper serverSocketWrapper = new ServerSocketWrapper();
+            ClientSocketWrapper clientSocketWrapper = new ClientSocketWrapper();
```
Comment (nadiacollado) [inline src/main/java/echoserver/EchoServer.java]: This causes the EchoServer to never interact with the EchoClient class directly and I think that may be where I'm probably setting this up incorrectly?

File: src/main/java/echoserver/EchoServer.java [hunk 3]
```diff
@@ -5,92 +5,37 @@
...
-        try {
-            input.close();
-            output.close();
-            clientSocket.close();
-            serverSocket.close();
         } catch (IOException e) {
-            e.printStackTrace();
+            System.out.println("Issue trying to listen on port " + port);
         }
+        serverSocket.close();
```
Comment (nadiacollado) [inline src/main/java/echoserver/EchoServer.java]: This never runs for some reason.

File: src/main/java/echoserver/ServerSocketWrapper.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        try {
+            clientSocket = serverSocket.accept();
+            System.out.println("Connection successful");
+        } catch (IOException e) {
+            System.out.println("Connection unsuccessful");
+        }
+        return clientSocket;
+    }
+
+    public void close() throws IOException {
```
Comment (nadiacollado) [inline src/main/java/echoserver/ServerSocketWrapper.java]: This should be called in the EchoServer but it says it's never used.

**Comment by wynspeare** [comment]
Typically you shouldn't commit the various `xml` and `.name` files that are generated by gradle/intelliJ. Take a look at https://www.gitignore.io. It will generate a comprehensive gitignore file for you based on the IDE and whatever build tools you are using.

File: build.gradle [hunk 5]
```diff
@@ -10,7 +10,27 @@ plugins {
...
+    // Use jcenter for resolving dependencies.
+    // You can declare any Maven/Ivy/file repository here.
+    mavenCentral()
+}
+
+dependencies {
+    implementation 'org.junit.jupiter:junit-jupiter:5.8.2'
+    implementation 'org.testng:testng:7.1.0'
+    implementation 'junit:junit:4.13.1'
+    testImplementation 'org.mockito:mockito-core:4.2.0'
```
Comment (wynspeare) [inline build.gradle]: I think maybe this mockito dependency can be removed?

**Comment by nadiacollado** [comment]
As is stands, I am only testing the EchoServer through the startEchoServer function which calls functions from the ServerSocketWrapper. I imagine that I need to be individually testing each function though? So wondering what that would like.

**Pull request merged**